Source/SwiftLintFramework/Rules/NoFallthroughRuleOnlyRule.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+import SourceKittenFramework
+
+public struct NoFallthroughOnlyRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "no_fallthrough_only",
+        name: "No Fallthrough Only",
+        description: "Fallthroughs can only be used if the `case` contains at least one other statement.",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            """
+            switch {
+            case 1:
+                var a = 1
+                fallthrough
+            case 2:
+                var a = 2
+            }
+            """,
+            """
+            switch {
+            case "a":
+                var one = 1
+                var two = 2
+                fallthrough
+            case "b": /* comment */
+                var three = 3
+            }
+            """,
+            """
+            switch {
+            case 1:
+               let one = 1
+            case 2:
+               // comment
+               var two = 2
+            }
+            """
+        ],
+        triggeringExamples: [
+            """
+            switch {
+            case 1:
+                fallthrough
+            case 2:
+                var a = 1
+            }
+            """,
+            """
+            switch {
+            case 1:
+                var a = 2
+            case 2:
+                fallthrough
+            case 3:
+                var a = 3
+            }
+            """,
+            """
+            switch {
+            case 1: // comment
+                fallthrough
+            }
+            """,
+            """
+            switch {
+            case 1: /* multi
+                line
+                comment */
+                fallthrough
+            case 2:
+                var a = 2
+            }
+            """
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: StatementKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        guard kind == StatementKind.case else {
+            return []
+        }
+
+        guard
+            let length = dictionary.length,
+            let offset = dictionary.offset
+        else {
+            return []
+        }
+
+        let pattern = "case[^:]+:\\s*" + // match start of case
```

Comment:
This rule could be more robust if we avoided regular expressions altogether, and instead operated on the dictionary that's being passed as a parameter in this function.

Reply:
It isn't clear to me how to use the dictionary to accomplish this. I am able to get the length and offset for the entire `case` block, but it doesn't seem to provide any additional breakdown. Do you have a suggestion for how to approach this?

Reply:
Sorry I wasn't clear. You should take a look at how `SwitchCaseAlignmentRule` is implemented.

For example, see how it filters the dictionary for case statements in a type-safe & AST-safe manner: https://github.com/realm/SwiftLint/blob/6a85b4348231cb5054961b05ded2f52a0d09ecc9/Source/SwiftLintFramework/Rules/SwitchCaseAlignmentRule.swift#L29-L32

Reply:
I'm not sure how this helps in my situation. The `case` statements exist as nodes on the dictionary, but it doesn't give me information about the individual tokens within the `case` statement. I tried grabbing the `offset` and `length` from the `case` node in the dictionary, and then used it to return the tokens in this range from the `SyntaxMap`. However, I can't think of an unambiguous way of filtering out the `case` keyword and any additional expressions before the colon.